packaging/amzn2/snapd.spec

```diff
@@ -0,0 +1,299 @@
+Group:          System Environment/Base
+Url:            https://%{provider_prefix}
+# Use the version with the vendored go packages
+Source0:  https://%{provider_prefix}/releases/download/%{version}/%{name}_%{version}.vendor.tar.xz
+BuildRequires:  autoconf
+BuildRequires:  automake
+BuildRequires:  gcc
+BuildRequires:  gettext
+BuildRequires:  gnupg
+BuildRequires:  pkgconfig(glib-2.0)
+BuildRequires:  pkgconfig(libcap)
+BuildRequires:  pkgconfig(libseccomp)
+BuildRequires:  pkgconfig(libudev)
+BuildRequires:  pkgconfig(systemd)
+BuildRequires:  libseccomp-static
+BuildRequires:  glibc-static
+BuildRequires:  golang
+BuildRequires:  libtool
+BuildRequires:  make
+BuildRequires:  pkgconfig
+BuildRequires:  squashfs-tools
+BuildRequires:  xfsprogs-devel
+BuildRequires:  xz
+BuildRequires:  valgrind
+
+%{?systemd_requires}
+Requires: squashfs-tools
+# TODO: enable squashfuse once it's packaged
+#Requires:  squashfuse
+#Requires:  fuse
+Requires: bash-completion
+
+BuildRoot:      %{_tmppath}/%{name}-%{version}-build
+
+%description
+This package contains that snapd daemon and the snap command line tool.
+Together they can be used to install, refresh (update), remove and configure
+snap packages on a system. Snap packages are a novel format based on simple
+principles. Bundle your dependencies, run in a predictable environment, use
+moder kernel features for setting up the execution environment and security.
+The same binary snap package can be installed and used on many diverse systems
+such as Debian, Fedora and OpenSUSE as well as their multiple derivatives.
+
+%prep
+%setup -q -n %{name}-%{version}
+
+%build
+# Set the version that is compiled into the various executables
+./mkversion.sh %{version}-%{release}
+
+# Build golang executables
+mkdir -p src/github.com/snapcore
+ln -s ../../../ src/github.com/snapcore/snapd
+
+export GOPATH=$(pwd):$(pwd)/Godeps/_workspace:%{gopath}
+
+GOFLAGS=
+%if 0%{?with_test_keys}
+GOFLAGS="$GOFLAGS -tags withtestkeys"
+%endif
+
+BUILDID="0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\\n')"
+
+%gobuild -o bin/snapd $GOFLAGS github.com/snapcore/snapd/cmd/snapd
+%gobuild -o bin/snap $GOFLAGS github.com/snapcore/snapd/cmd/snap
+%gobuild -o bin/snapctl $GOFLAGS github.com/snapcore/snapd/cmd/snapctl
+%gobuild_static -o bin/snap-exec $GOFLAGS github.com/snapcore/snapd/cmd/snap-exec
+%gobuild_static -o bin/snap-update-ns $GOFLAGS github.com/snapcore/snapd/cmd/snap-update-ns
+%gobuild -o bin/snap-seccomp $GOFLAGS github.com/snapcore/snapd/cmd/snap-seccomp
+
+# Build C executables
+
+# Generate autotools build system files
+pushd cmd
+autoreconf -i -f
+
+%configure  \
+    --disable-apparmor \
+    --libexecdir=/usr/libexec/snapd \
+    --with-snap-mount-dir=%{_sharedstatedir}/snapd/snap \
+    --with-merged-usr
+
+%make_build
+popd
+
+pushd ./data
+# Build data
+make BINDIR="%{_bindir}" LIBEXECDIR="%{_libexecdir}" \
+     SYSTEMDSYSTEMUNITDIR="%{_unitdir}" \
+     SNAP_MOUNT_DIR="%{_sharedstatedir}/snapd/snap" \
```

Comment:
Let's detect amazon linux in snapd and use /snap here please.

Reply:
I am ok with using `/snap`. This require some extra changes in our runtime distro detection. For the record the contents of `/etc/os-release`:

```
[ec2-user@localhost ~]$ cat /etc/os-release 
NAME="Amazon Linux"
VERSION="2 (2017.12)"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2 (2017.12) LTS Release Candidate"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
```

Reply:
@bboozzoo Actually, it wouldn't, because if the `/snap` symlink exists, it will work.